packages/openid4vc/src/openid4vc-issuer/repository/OpenId4VcIssuanceSessionRecord.ts

```diff
@@ -1,39 +1,104 @@
 import type { OpenId4VciCredentialOfferPayload } from '../../shared'
 import type { RecordTags, TagsBase } from '@credo-ts/core'
 
-import { CredoError, BaseRecord, utils, DateTransformer } from '@credo-ts/core'
+import { PkceCodeChallengeMethod } from '@animo-id/oauth2'
+import { CredoError, BaseRecord, utils } from '@credo-ts/core'
 import { Transform } from 'class-transformer'
 
 import { OpenId4VcIssuanceSessionState } from '../OpenId4VcIssuanceSessionState'
 
 export type OpenId4VcIssuanceSessionRecordTags = RecordTags<OpenId4VcIssuanceSessionRecord>
 
+export interface OpenId4VcIssuanceSessionAuthorization {
+  code?: string
+
+  /**
+   * @todo: I saw in google's library that for codes they encrypt an id with expiration time.
+   * You now the code was created by you because you can decrypt it, and you don't have to store
+   * additional metadata on your server. It's similar to the signed / encrypted nonce
+   */
+  codeExpiresAt?: Date
+
+  /**
+   * String value created by the Credential Issuer and opaque to the Wallet that
+   * is used to bind the subsequent Authorization Request with the Credential Issuer to a context set up during previous steps.
+   */
+  issuerState?: string
+}
+
+export interface OpenId4VcIssuanceSessionPresentation {
+  /**
+   * Whether presentation during issuance is required.
+   */
+  required: true
```

Comment:
should this not be a `boolean`?

Reply:
The object will never be defined it it's not true. So i think `true` is more clear here. Alternative is empty object to indicate it's required but that felt not explicit.

Or another top-level prop for `presentationRequired` and then `presentation` for the details, but then you can still have that `presentation` is undefined (in TS) even if `required` is true. So no if you check for `required == true`, the other props will also be defined